src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -518,11 +518,13 @@ export class SearchSource {
       );
       const uniqFieldNames = [...new Set([...bodyFieldNames, ...fieldsFromSource])];
 
-      // filter down script_fields to only include items specified
-      body.script_fields = pick(
-        body.script_fields,
-        Object.keys(body.script_fields).filter((f) => uniqFieldNames.includes(f))
-      );
+      if (uniqFieldNames.indexOf('*') === -1) {
```

Comment:
:art: Use ES2015+ syntax

```suggestion
      if (!uniqFieldNames.includes('*')) {
```